packages/aws-cdk/lib/api/hotswap/monitor/logs-monitor.ts

```diff
@@ -0,0 +1,226 @@
+      clearTimeout(this.tickTimer);
+    }
+  }
+
+  private scheduleNextTick() {
+    if (!this.active) {
+      return;
+    }
+    this.tickTimer = setTimeout(() => void (this.tick()), this.printer.updateSleep);
+  }
+
+  private async tick() {
+    if (!this.active) {
+      return;
+    }
+    try {
+      this.readPromise = this.readNewEvents();
+      await this.readPromise;
+      this.readPromise = undefined;
+
+      if (!this.active) { return; }
+
+    } catch (e) {
+      error('Error occurred while monitoring logs: %s', e);
+    }
+    this.scheduleNextTick();
+
+  }
+
+  /**
+   * Reads all new log events from a set of CloudWatch Log Groups
+   * in parallel
+   */
+  private async readNewEvents(): Promise<void[]> {
+    const promises: Array<Promise<void>> = [];
+    for (let group of this.logGroups) {
+      promises.push(this.readEventsFromLogGroup(group));
+    }
+    return Promise.all(promises);
+  }
+
+  /**
+   * Reads all new log events from a CloudWatch Log Group
+   * starting when the hotswap was triggered
+   *
+   * Only prints out events that have not been printed already
+   */
+  private async readEventsFromLogGroup(logGroupName: string): Promise<void> {
+    // this should not be possible
+    if (!this.sdk) {
+      return;
+    }
+    try {
+      let finished = false;
+      let nextToken: string | undefined;
+      while (!finished) {
+        const response = await this.sdk!.cloudwatchLogs().filterLogEvents({
+          logGroupName: logGroupName,
+          nextToken,
+          startTime: this.startTime,
+        }).promise();
+        const eventPage = response.events ?? [];
+
+
+        for (const event of eventPage) {
+          if (event.eventId && this.activity.has(event.eventId)) {
+          } else {
+            if (event.eventId) this.activity.add(event.eventId);
+            if (event.message) {
+              const ingestionTime: Date = event.ingestionTime ?
+                new Date(event.ingestionTime) : new Date();
+              this.printer.print({
+                message: event.message,
+                logGroup: logGroupName,
+                logStream: event.logStreamName,
+                ingestionTime,
+              });
+            }
+
+          }
+        }
+
+        nextToken = response.nextToken;
+        if (nextToken === undefined) {
+          finished = true;
+        }
+      }
+    } catch (e) {
+      // with Lambda functions the CloudWatch is not created
+      // until something is logged, so just keep polling until
+      // there is somthing to find
+      if (e.code === 'ResourceNotFoundException') {
+        return;
+      }
+      throw e;
+    }
+  }
+
+}
+
+interface PrinterProps {
+  /**
+   * Stream to write to
+   */
+  readonly stream: NodeJS.WriteStream;
+}
+
+export interface IEventPrinter {
+  readonly updateSleep: number;
+
+  print(event: CloudWatchLogEvent): void;
+}
+
+abstract class EventPrinterBase {
+  public readonly updateSleep: number = 5_000;
```

Comment:
1. Let's move this to a file-level, global value - this can't be changed anyway right now, and so doesn't really belong as an instance property of a class.
2. Is 5 seconds not a little too slow? Any chance we can ramp this up to something like 2? Right now, the delay is pretty noticeable when using the option.

Reply:
Updated to a global value and set to 2 seconds.